### PR TITLE
feat(cblog): add finaliser and comprehensive test coverage

### PR DESCRIPTION
**Pull request by amery**
## Summary

- Add finaliser to automatically close internally created channels using `runtime.SetFinalizer`
- Add comprehensive test coverage for cblog handler achieving 100% coverage
- Implement full slog interface compliance testing
- Add worker pattern testing for callback functionality

### Core Feature (cblog.go)
- **Finaliser support**: When `cblog.New(nil)` creates a channel internally, set a finaliser to automatically close it when the logger is garbage collected
- Prevents potential channel leaks for internally created channels
- External channels remain unaffected

### Test Coverage (3 new test files)
- **cblog_test.go** (557 lines): Core functionality testing including concurrency, field handling, and edge cases
- **compliance_test.go** (150 lines): Full slog interface compliance testing using the shared testing framework
- **worker_test.go** (168 lines): Callback functionality, worker patterns, and concurrent message handling

### Coverage Results
- cblog handler: **100.0%** statement coverage
- All other handlers maintain existing coverage levels
- All tests pass including race condition testing

## Test plan

- [x] All tests pass: `make test`
- [x] Full coverage maintained: `make coverage` 
- [x] Linting passes: `make tidy`
- [x] Compliance testing verifies full slog interface implementation
- [x] Concurrency testing validates thread safety
- [x] Memory management testing for finaliser functionality

## Summary by CodeRabbit

* **Bug Fixes**
  * Improved automatic cleanup of internal resources for the logging handler, ensuring channels created internally are properly closed when no longer needed.

* **Tests**
  * Added comprehensive unit, concurrency, and compliance tests for the logging handler, covering message formatting, field handling, logger levels, stack traces, and concurrent usage.
  * Introduced tests for message ordering, blocking behavior, and performance benchmarks to validate handler reliability and efficiency.

* **Chores**
  * Updated spell checker dictionary to recognize additional terms.

### CODE REVIEW

File: handlers/cblog/cblog.go [hunk 1]
```diff
@@ -169,11 +170,21 @@ func (l *Logger) WithFields(fields map[string]any) slog.Logger {
...
 
 	l := newLogger(ch)
+
+	// Set finaliser to close the channel if we created it
+	if createdChannel {
+		runtime.SetFinalizer(l, func(l *cblog) {
+			close(l.ch)
+		})
+	}
+
```
[the remark this answers is not in the document]
Comment (amery) [reply handlers/cblog/cblog.go:187]: if the cblog is being GC'ed, sendMsg shouldn't be callable

File: handlers/cblog/cblog_test.go [hunk 2]
```diff
@@ -0,0 +1,558 @@
...
+		// Drain channel in background
+		go func() {
+			var count int
+			for range ch {
+				count++
+			}
+		}()
+		return logger
+	})
+}
```
[the remark this answers is not in the document]
Comment (amery) [reply handlers/cblog/cblog_test.go:421]: check now please
